pyccl/pk2d.py

```diff
@@ -149,3 +209,43 @@ def __del__(self):
         if hasattr(self, 'has_psp'):
             if self.has_psp and hasattr(self, 'psp'):
                 lib.f2d_t_free(self.psp)
+
+
+def parse_pk2d(cosmo, p_of_k_a, is_linear=False):
+    """ Return the C-level `f2d` spline associated with a
+    :class:`Pk2D` object.
+
+    Args:
+        cosmo (:class:`~pyccl.core.Cosmology`): A Cosmology object.
+        p_of_k_a (:class:`Pk2D`, :obj:`str` or `None`): if a
+            :class:`Pk2D` object, its `f2d` spline will be used. If
+            a string, the linear or non-linear power spectrum stored
+            by `cosmo` under this name will be used. If `None`, the
+            matter power spectrum stored by `cosmo` will be used.
+        is_linear (:obj:`bool`): if `True`, and if `p_of_k_a` is a
+            string or `None`, the linear version of the corresponding
+            power spectrum will be used (otherwise it'll be the
+            non-linear version).
+    """
+    if isinstance(p_of_k_a, Pk2D):
+        psp = p_of_k_a.psp
+    else:
+        if isinstance(p_of_k_a, str):
+            name = p_of_k_a
+        elif p_of_k_a is None:
+            name = 'delta_matter_x_delta_matter'
+        else:
+            raise ValueError("p_of_k_a must be a pyccl.Pk2D object, "
+                             "a string, or None")
+
+        if is_linear:
+            cosmo.compute_linear_power()
+            pk = cosmo._pk_lin
+        else:
+            cosmo.compute_nonlin_power()
+            pk = cosmo._pk_nl
```

Comment:
I suspect we need an additional interface on the cosmology object that looks like `get_linear_power` and `get_nonlin_power` that pulls these attributes from the dicts and calls the compute functions if needed.

Reply:
Humm, I wrote evaluators [here](https://github.com/LSSTDESC/CCL/blob/6dcd05f7c3990dd7706f58d6996b1c86e3965639/pyccl/power.py#L8) and [here](https://github.com/LSSTDESC/CCL/blob/6dcd05f7c3990dd7706f58d6996b1c86e3965639/pyccl/power.py#L28). Do you think we need something that returns the Pk2D object itself? Not sure if that'd be needed too often.

Reply:
On second thought, I see what you mean. Let me write those.

Reply:
Ok, I've created them [here](https://github.com/LSSTDESC/CCL/blob/e8883f969a67809c38381fffed47391f54280024/pyccl/core.py#L874) and [here](https://github.com/LSSTDESC/CCL/blob/e8883f969a67809c38381fffed47391f54280024/pyccl/core.py#L893).